src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalNamedFormula.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Entities.Delegation;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    internal interface IExternalNamedFormula : IExternalPageableSymbol, IExternalDelegable
+    {
+        bool TryGetExternalDataSource(out IExternalDataSource dataSource);
```

Comment:
Should this method just be on IExternalDelegatableSymbol?

Reply:
It's very specific to named formula. Hence it's on IExternalNamedFormula. If it's needed for other entities then we can think about moving it there.